File: src/main/java/red/tetracube/iot/smartigloohub/data/entities/Environment.java
package red.tetracube.iot.smartigloohub.data.entities;

import java.util.Collections;
import java.util.List;
import java.util.UUID;
import javax.persistence.*;

@Entity
@Table(schema = "igloo_hub", name = "environments")
public class Environment {

    @Id
    private UUID id;

    @Column(name = "name", nullable = false)
    private String name;

    @OneToMany(fetch = FetchType.LAZY, targetEntity = Switcher.class, mappedBy = "environment")
    private List<Switcher> environments = Collections.emptyList();

}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/data/entities/Environment.kt
package red.tetracube.iot.smartigloohub.data.entities

import java.util.UUID
import javax.persistence.*

@Entity
@Table(schema = "igloo_hub", name = "environments")
data class Environment(
    @Id
    val id: UUID? = null,

    @Column(name = "name", nullable = false)
    val name: String? = null,

    @JoinColumn(name = "house_id", nullable = false)
    @ManyToOne(targetEntity = House::class, fetch = FetchType.LAZY)
    val house: House? = null,

    @OneToMany(fetch = FetchType.LAZY, targetEntity = Switcher::class, mappedBy = "environment")
    val environments: List<Switcher> = emptyList()
)
